src/System.IO.FileSystem.Watcher/src/System/IO/FileSystemWatcher.OSX.cs

```diff
@@ -295,7 +247,7 @@ private void WatchForFileSystemEventsThreadStart()
                 else if ((DoesPathPassNameFilter(eventPaths[i])) && ((_filterFlags & eventFlags[i]) != 0))
                 {
                     // Remove the base directory prefix (including trailing / that OS X adds)
-                    string relativePath = eventPaths[i].Remove(0, _fullDirectory.Length + 1);
+                    string relativePath = (eventPaths[i].Equals(_fullDirectory, StringComparison.CurrentCultureIgnoreCase) ? _fullDirectory : eventPaths[i].Remove(0, _fullDirectory.Length).TrimStart('/'));
```

Comment:
Rather than doing a Remove and then a Trim, can we just manually count how many '/'s there are after _fullDirectory.Length and pass that value as the length to Remove?  As it is we're potentially creating two strings when we could instead just do one.